ResearchKit/Charts/ORKGraphChartView.h

```diff
@@ -229,9 +218,66 @@ ORK_AVAILABLE_DECL
 
 
 /**
- The `ORKGraphChartView` class is an abstract class. It holds properties and methods common to
- subclasses such as `ORKLineGraphChartView` and `ORKDiscreteGraphChartView`. You should not instantiate
- this class directly; use one of the subclasses instead.
+ An object that adopts the `ORKValueRangeGraphChartViewDataSource` protocol is responsible for
+ providing data in the form of `ORKValueRange` values required to populate an
+ `ORKValueRangeGraphChartView` concrete subclass, such as `ORKLineGraphChartView` and
+ `ORKDiscreteGraphChartView`.
+ */
+ORK_AVAILABLE_DECL
+@protocol ORKValueRangeGraphChartViewDataSource <ORKGraphChartViewDataSource>
+
+@required
+
+/**
+ Asks the data source for the value range to be plotted at the specified point index for the
+ specified plot.
+ 
+ @param graphChartView      The graph chart view that is asking for the value range.
+ @param pointIndex          An index number identifying the range point in the graph chart view.
+ @param plotIndex           An index number identifying the plot in the graph chart view. This index
+                                is 0 in a single-plot graph chart view.
+ 
+ @return The value range specified by `pointIndex` in the plot specified by `plotIndex` for the
+ specified graph chart view`.
+ */
+- (ORKValueRange *)graphChartView:(ORKGraphChartView *)graphChartView pointForPointIndex:(NSInteger)pointIndex plotIndex:(NSInteger)plotIndex;
+
+@end
+
+
+/**
+ An object that adopts the `ORKValueStackGraphChartViewDataSource` protocol is responsible for
+ providing data in the form of `ORKValueStack` values required to populate an `ORKBarGraphChartView`
+ object.
+ */
+ORK_AVAILABLE_DECL
+@protocol ORKValueStackGraphChartViewDataSource <ORKGraphChartViewDataSource>
+
+@required
+
+/**
+ Asks the data source for the value stack to be plotted at the specified point index for the
+ specified plot.
+ 
+ @param graphChartView      The graph chart view that is asking for the range point.
+ @param pointIndex          An index number identifying the range point in the graph chart view.
+ @param plotIndex           An index number identifying the plot in the graph chart view. This index
+ is 0 in a single-plot graph chart view.
+ 
+ @return The value stack specified by `pointIndex` in the plot specified by `plotIndex` for the
+ specified graph chart view`.
+ */
+- (ORKValueStack *)graphChartView:(ORKGraphChartView *)graphChartView pointForPointIndex:(NSInteger)pointIndex plotIndex:(NSInteger)plotIndex;
```

Comment:
I agree that `pointForPointIndex` is suboptimal.

But we would have the same problem as above, it's not clear if value is a *value collection class* or an *individual double value*. Ah! Difficult naming conventions! :stuck_out_tongue_winking_eye: